lib/message.ex

```diff
@@ -309,11 +310,23 @@ defmodule LangChain.Message do
       do: true
 
   def is_function_call?(%Message{}), do: false
+
+  def add_image(%Message{role: :user, images: images} = message, image) do
+    %Message{
+      message | images: images ++ [image]
+    }
+  end
+
+  def has_images?(%Message{images: images}) do
+    length(images) > 0
+  end
```

Comment:
This would be better as two functions and a pattern match.

```elixir
def has_images?(%Message{images: []}), do: false

def has_images?(%Message{images: images}) when is_list(images), do: true
```

This is more explicit and avoids traversing the full list just to know if it's empty or not.